src/components/button/button.d.ts

```diff
@@ -1,15 +1,22 @@
 import * as React from 'react';
+
 export interface ButtonProps {
   as?: 'primary' | 'secondary' | 'tertiary' | 'dashed' | 'destructive' | 'darkBackground';
+  buttonType?: 'primary' | 'secondary' | 'tertiary' | 'dashed' | 'destructive' | 'darkBackground';
   disabled?: boolean;
+  destructive?: boolean;
   size?: 'small' | 'medium' | 'large';
+  iconPosition?: 'before' | 'after';
+  iconType?: 'add' | 'alert' | 'analysis' | 'arrow_down' | 'arrow_left' | 'arrow_left_boxed' | 'arrow_left_small' | 'arrow_right' | 'arrow_right_small' | 'arrow_up' | 'attach' | 'bank' | 'basket' | 'basket_with_squares' | 'bin' | 'blocked' | 'blocked_square' | 'block_arrow_right' | 'boxed_shapes' | 'bulk_destroy' | 'business' | 'calendar' | 'calendar_today' | 'call' | 'camera' | 'card_view' | 'caret_down' | 'cart' | 'chat' | 'chart_bar' | 'chart_line' | 'chart_pie' | 'chat_notes' | 'chevron_down' | 'chevron_left' | 'chevron_right' | 'chevron_up' | 'circles_connection' | 'clock' | 'close' | 'coins' | 'collaborate' | 'copy' | 'connect' | 'credit_card' | 'credit_card_slash' | 'cross' | 'csv' | 'delete' | 'delivery' | 'disputed' | 'document_right_align' | 'document_tick' | 'document_vertical_lines' | 'download' | 'drag' | 'drag_vertical' | 'draft' | 'dropdown' | 'duplicate' | 'edit' | 'edited' | 'email' | 'ellipsis_horizontal' | 'ellipsis_vertical' | 'error' | 'error_square' | 'factory' | 'favourite' | 'favourite_lined' | 'fax' | 'feedback' | 'files_leaning' | 'filter' | 'filter_new' | 'fit_height' | 'fit_width' | 'folder' | 'gift' | 'graph' | 'grid' | 'help' | 'home' | 'image' | 'in_progress' | 'in_transit' | 'info' | 'individual' | 'key' | 'ledger' | 'ledger_arrow_left' | 'ledger_arrow_right' | 'link' | 'list_view' | 'locked' | 'location' | 'logout' | 'lookup' | 'marker' | 'message' | 'messages' | 'minus' | 'minus_large' | 'mobile' | 'money_bag' | 'pause_circle' | 'pdf' | 'people' | 'person' | 'person_tick' | 'phone' | 'play' | 'play_circle' | 'plus' | 'plus_large' | 'print' | 'progress' | 'progressed' | 'question' | 'refresh' | 'refresh_clock' | 'remove' | 'save' | 'scan' | 'search' | 'services' | 'settings' | 'share' | 'shop' | 'sort_down' | 'sort_up' | 'spanner' | 'split' | 'split_container' | 'stacked_boxes' | 'stacked_squares' | 'submitted' | 'sync' | 'tag' | 'three_boxes' | 'tick' | 'unlocked' | 'upload' | 'uploaded' | 'video' | 'view' | 'warning';
```

Comment:
You should probably use our existing `IconTypes` type here
src/utils/helpers/options-helper/options-helper.d.ts

Reply:
I was looking at creating my own version of this, I missed that there was already an IconTypes declared.